### DISCUSSION TITLE
Deleting Files

### DISCUSSION THREAD

**Discussion by nateProjects** [micropython/micropython#13321] [Using MicroPython]
Hiya! Why doesn’t this work (it says it does)?

```
import os
config = "state.json"
try:
    os.remove(config)
    print("Config deleted successfully")
except OSError:
    print("Error deleting config")  # Handle potential errors
```

I’ve checked the path (/) and filename.
& I’ve tried import uos / uos.remove(config) as well as os.unlink(config) too.

What am I missing? Any ideas?

_Note: This is on a Pico W / Inky Frame_

**Comment by robert-hh** [comment]
I am pretty sure that os.remove() works. I use it regularly. If you see that behaviour using Thonny, try the same without Thonny. You can as well use os.listdir() to get the „raw“ file list not processed by any IDE.

**Comment by GitHubsSilverBullet** [comment]
```python
import os
config = "state.json"
try:
    os.remove(config)
    print("Config deleted successfully")
except OSError as e:
    print(e)
```
What kind of OSError do you get?
